sources/topology_optimization/bkp_design.f90

```diff
@@ -0,0 +1,347 @@
+!> @file topology_module.f90
+!! @brief Module containing the topology type
+!! @details This module contains the topology type, which is used to describe
+!!          the designs in topology optimization.
+
+module design_module
+
+  use case, only: case_t
+  use fluid_user_source_term, only: fluid_user_source_term_t
+  ! use fluid_source_term, only: fluid_source_term_t
+  use field, only: field_t
+  use field_registry, only: neko_field_registry
+  use num_types, only: rp
+  use point_zone, only: point_zone_t
+  use device, only: device_map, device_memcpy, HOST_TO_DEVICE
+  use logger, only: neko_log
+  use point_zone_registry, only: neko_point_zone_registry
+  use json_utils, only: json_get, json_get_or_default
+  use neko_config, only: NEKO_BCKND_DEVICE
+  use math, only: col3, rzero, col2, cmult, invcol1, cadd, add2, cfill
+  use device_math, only: device_col3, device_rzero, device_col2, device_cmult, &
+       device_invcol1, device_cadd, device_add2, device_cfill
+  use math_ext, only: cadd_mask, col3_mask
+  use device_math_ext, only: device_cadd_mask, device_col3_mask
+  use simulation_component, only: simulation_component_t
+  use json_module, only: json_file
+  use utils, only: neko_error
+  use, intrinsic :: iso_c_binding, only: c_ptr, c_null_ptr
+  ! use topopt_brinkman_source_term, only: topopt_brinkman_source_term_t
+  implicit none
+
+
+  ! ========================================================================= !
+  ! Module interface
+  ! ========================================================================= !
+  private
+
+  public :: design_t, topopt_permeability_force
+
+  ! ========================================================================= !
+  ! Global variables
+  ! ========================================================================= !
+
+  !> @brief The resistance array
+  real(kind=rp), dimension(:), allocatable :: resistance
+  !> @brief Pointer to the resistance array on the device
+  type(c_ptr) :: resistance_d = c_null_ptr
+  !> @brief Pointer to the design domain mask
+  integer, dimension(:), pointer :: design_domain_mask
+  !> @brief Pointer to the design domain mask on the device
+  type(c_ptr) :: design_domain_mask_d = c_null_ptr
+
+  ! ========================================================================= !
+  ! Topology type
+  ! ========================================================================= !
+
+  !> @brief The topology type, which is used to describe the designs in
+  !! topology optimization
+  type, extends(simulation_component_t) :: design_t
```

Comment:
This should not be a simulation component.

Reply:
that was a backup of the pre-existing design that was already there. we can delete it.